### PR TITLE
LazyRelayInReq: read and carry headers and endpoint

### PR DESCRIPTION
**Pull request by jcorbin**
r @Raynos @kriskowal

### CODE REVIEW

File: relay_handler.js [hunk 1]
```diff
@@ -165,21 +167,31 @@ LazyRelayInReq.prototype.initRead =
...
         return res.err;
     }
     self.serviceName = res.value;
 
-    // TODO: lazy read self.callerName
+    res = self.reqFrame.bodyRW.lazy.readHeaders(self.reqFrame);
+    if (res.err) {
+        return res.err;
+    }
+    self.headers = res.value;
```
Comment (Raynos) [inline relay_handler.js]: `self.callerName = self.headers.cn` ?

File: relay_handler.js [hunk 2]
```diff
@@ -205,6 +217,8 @@ function _extendLogInfo(info) {
     info.inRemoteAddr = self.remoteAddr;
     info.inRequestId = self.id;
     info.serviceName = self.serviceName;
+    info.callerName = self.callerName;
```
Comment (Raynos) [inline relay_handler.js]: no `self.callerName` yet. always undefined

File: relay_handler.js (hunk 1 above)
Comment (jcorbin) [inline relay_handler.js]: good catch, but that's not how this headers object works, see lazyRead in v2/header for more; fix pushed to this branch to actually use `self.headers.getValue(Buffer('cn'))`

File: relay_handler.js (hunk 2 above)
Comment (jcorbin) [inline relay_handler.js]: empty string, not undefined, but valid point still :-)

File: relay_handler.js [hunk 3]
```diff
@@ -28,6 +28,9 @@ RelayHandler.RelayRequest = RelayRequest;
 
 module.exports = RelayHandler;
 
+// "constant" byte buffer used for lookup in LazyRelayInReq#initRead
+var cnBytes = Buffer('cn');
```
Comment (Raynos) [inline relay_handler.js]: nitpick: `new Buffer('cn');`

**Comment by Raynos** [comment]
lgtm.

**Pull request merged**